src/com/owncloud/android/ui/activity/FileDisplayActivity.java

```diff
@@ -802,6 +804,12 @@ protected void onResume() {
         super.onResume();
         Log_OC.e(TAG, "onResume() start");
         
+        if (PinCheck.checkIfPinEntry()){
+            Intent i = new Intent(MainApp.getAppContext(), PinCodeActivity.class);
```

Comment:
For the record: onStart() / onStop() will not do it; when you go back from an activity to the previous one, onStart() in the older is called before onStop() in the exiting one; the time that will compared in onStart() is with its own stop before starting the newest.